telluric/vrt.py

```diff
@@ -136,4 +152,87 @@ def boundless_vrt_doc(
         dst_window = Window(xoff, yoff, xsize, ysize)
         vrt.add_band_simplesource(mask_band, 'mask,1', 'Byte', False, src_dataset.name,
                                   width, height, block_shape[1], block_shape[0], src_window, dst_window)
-    return vrt.tostring()
+    return vrt
+
+
+def band_name_to_color_interpretation(band_name):
+    if band_name.lower() in ['red', 'green', 'blue']:
+        return band_name
+    if band_name.lower().endswith('_enhanced'):
+        return band_name_to_color_interpretation(band_name[:-len('_enhanced')])
+    else:
+        return 'grey'
+
+
+def raster_list_vrt(rasters, relative_to_vrt=True):
+    """Make a VRT XML document from a list of GeoRaster2 objects.
+    Parameters
+    ----------
+    rasters : list
+        The list of GeoRasters.
+    relative_to_vrt : bool, optional
+        If True the bands simple source url will be related to the VRT file
+    Returns
+    -------
+    bytes
+        An ascii-encoded string (an ElementTree detail)
+    """
+
+    from telluric import FeatureCollection
+    fc = FeatureCollection.from_georasters(rasters)
+    return raster_collection_vrt(fc, relative_to_vrt)
+
+
+def raster_collection_vrt(fc, relative_to_vrt=True):
+    """Make a VRT XML document from a feature collection of GeoRaster2 objects.
+    Parameters
+    ----------
+    rasters : FeatureCollection
+        The FeatureCollection of GeoRasters.
+    relative_to_vrt : bool, optional
+        If True the bands simple source url will be related to the VRT file
+    Returns
+    -------
+    bytes
+        An ascii-encoded string (an ElementTree detail)
+    """
+
+    def max_resolution():
+        max_affine = max(fc, key=lambda f: f.raster.resolution()).raster.affine
+        return abs(max_affine.a), abs(max_affine.e)
+
+    from telluric import rasterization
+    assert all(fc.crs == f.crs for f in fc), "all rasters should have the same CRS"
+
+    rasters = (f.raster for f in fc)
+    bounds = fc.convex_hull.get_bounds(fc.crs)
+    resolution = max_resolution()
+    width, height, affine = rasterization.raster_data(bounds, resolution)
+
+    bands = {}  # type: Dict[str, tuple]
+    vrt = BaseVRT(width, height, fc.crs, affine)
+
+    last_band_idx = 0
+    for raster in rasters:
+        for i, band_name in enumerate(raster.band_names):
+            if band_name in bands:
+                band_element, band_idx = bands[band_name]
+            else:
+                last_band_idx += 1
+                band_idx = last_band_idx
+                band_element = vrt.add_band(raster.dtype, band_idx, band_name_to_color_interpretation(band_name))
+                bands[band_name] = (band_element, last_band_idx)
+
+            src_window = Window(0, 0, raster.width, raster.height)
+            xoff = (raster.affine.xoff - affine.xoff) / affine.a
```

Comment:
is there a way to use https://rasterio.readthedocs.io/en/latest/api/rasterio.windows.html?highlight=window#rasterio.windows.WindowMethodsMixin.window in order to caclulate dest_window?

In the case we had to calculate the window out of location we used this method, can we continue do it?

Reply:
I did not find there a way to place a window within another window based on their real world location(affine)